hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/ServerCrashProcedure.java

```diff
@@ -488,6 +496,8 @@ protected boolean isMatchingRegionLocation(RegionStateNode rsn) {
    */
   private void assignRegions(MasterProcedureEnv env, List<RegionInfo> regions) throws IOException {
     AssignmentManager am = env.getMasterServices().getAssignmentManager();
+    boolean retainAssignment = env.getMasterConfiguration().getBoolean(SCP_RETAIN_ASSIGNMENT,
```

Comment:
we should also check whether the current servername is online.
like "isServerWithSameHostnamePortOnline"